content/300-guides/300-upgrade-guides/200-upgrading-versions/030-upgrading-to-prisma-4.mdx

```diff
@@ -0,0 +1,418 @@
+
+- upgrade to the new Prisma 4 packages following [these instructions](#upgrade-the-prisma-and-prismaclient-packages-to-prisma-4)
+- manually fix the validation errors in your Prisma schema. Alternatively, if you have an up-to-date live database, running `prisma db pull` will add the `@unique` attributes automatically.
+
+### Disallow `references` syntax for implicit many-to-many relations
+
+When using [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) in Prisma 4, you will no longer be able to use the `references` argument, which was previously optional. For example, the following relation would now trigger a validation error:
+
+```prisma file=schema.prisma
+model Post {
+  id         Int        @id @default(autoincrement())
+  categories Category[] @relation("my-relation", references: [id]) // <-- validation error
+}
+
+model Category {
+  id    Int    @id @default(autoincrement())
+  posts Post[] @relation("my-relation", references: [id]) // <-- validation error
+}
+```
+
+Instead, you can write:
+
+```prisma file=schema.prisma
+model Post {
+  id         Int        @id @default(autoincrement())
+  categories Category[] @relation("my-relation")
+}
+
+model Category {
+  id    Int    @id @default(autoincrement())
+  posts Post[] @relation("my-relation")
+}
+```
+
+This is because the only valid value for `references` was `id`, so removing this argument makes it clearer what can and cannot be changed.
+
+#### Upgrade path
+
+After you upgrade to Prisma 4, any implicit many-to-many relations with a `references` argument will trigger a validation error. To upgrade, you will need to:
+
+- upgrade to the new Prisma 4 packages following [these instructions](#upgrade-the-prisma-and-prismaclient-packages-to-prisma-4)
+- manually fix the validation errors in your Prisma schema. Alternatively, if you have an up-to-date live database, running `prisma db pull` will remove the `references` arguments automatically.
+
+### Better grammar for string literals
+
+String literals in your Prisma Schema now need to follow the same rules as strings in JSON. This mostly changes the escaping of some special characters. More details can be found in [the JSON specification](https://www.ietf.org/rfc/rfc4627.txt) or on the [JSON website](https://www.json.org/json-en.html).
+
+#### Upgrade path
+
+This is a breaking change for some existing schemas. After you upgrade to Prisma 4, incorrectly escaped characters will trigger a validation error. To upgrade, you will need to:
+
+- upgrade to the new Prisma 4 packages following [these instructions](#upgrade-the-prisma-and-prismaclient-packages-to-prisma-4)
+- manually fix the validation errors in your Prisma schema.
+
+## Upgrade the <inlinecode>prisma</inlinecode> and <inlinecode>@prisma/client</inlinecode> packages to Prisma 4
```

Comment:
I wonder if this should be higher up in the document? It feels like a "how to upgrade" section.